googleapi/render.js

```diff
@@ -0,0 +1,100 @@
+const electron = require('electron');
+const ipcRenderer = electron.ipcRenderer;
+
+const constants = require('./constants');
+
+/**
+ * A private reference to the iframe with the Jitsi Meet meeting.
+ *
+ * @private
+ * @type {InlineFrameElement}
+ */
+let iframe = null;
+
+/**
+ * A helper object for performing interactions with the Google API that may not
+ * be possible otherwise in Electron with the Google JS Client.
+ *
+ * @type {Object}
+ */
+const googleApi = {
+    /**
+     * Sends a request to the main process to open the passed in Google auth
+     * URL and sets a listener for the main process's response.
+     *
+     * @param {Object} options - An object describing how to handle the auth
+     * flow.
+     * @param {string} options.authUrl - The Google oauth URL to visit to ask
+     * for access to the user's account.
+     * @param {string} options.rediredirectUrl - The URL that is expected to be
+     * visited after the Google oauth process is complete.
+     * @returns {Promise}
+     */
+    requestToken(options) {
+        return new Promise(resolve => {
+            ipcRenderer.once(constants.AUTH_FINISHED, (event, url) => {
+                resolve(url);
+            });
+
+            ipcRenderer.send(constants.AUTH_REQUESTED, options);
+        });
+    }
+};
+
+/**
+ * Sets {@code googleApi} on a global variable for Jitsi Meet to access.
+ *
+ * @private
+ * @returns {void}
+ */
+function setGlobalGoogleApiVariable() {
+    if (!iframe) {
+        return;
+    }
+
+    iframe.contentWindow.JitsiMeetElectron
+        = iframe.contentWindow.JitsiMeetElectron || {};
+
+    iframe.contentWindow.JitsiMeetElectron.googleApi = googleApi;
+}
+
+/**
+ * Sets a global variable on the Jitsi Meet iFrame to expose Google API
+ * functionality that needs to be executed in Electron's main context.
+ *
+ * @param {JitsiIFrameApi} api - The Jitsi Meet iframe api object.
+ * @returns {void}
+ */
+function setupGoogleApiRender(api) {
+    teardownGoogleApiRender();
+
+    iframe = api.getIFrame();
+
+    iframe && iframe.addEventListener('load', setGlobalGoogleApiVariable);
+}
+
+/**
+ * Removes listeners that may have been set for interacting with the Google API.
+ *
+ * @returns {void}
+ */
+function teardownGoogleApiRender() {
+    if (iframe) {
```

Comment:
What do you think about the idea to replace the attaching of the method to the iframe window's DOM with a "private" listener on the iframe API?

IMHO using the iframe API vs directly manipulating the DOM looks cleaner to me. Also in this case the whole logic in the render process is to pass and receive IPC messages and real implementation is in the main process.

Reply:
Okay, so take in the api and add a listener through some kind of api.addListener method? Does that method exist already? I'd want to change the screenshare setup as well then because I re-used its approach.

Reply:
So you can probably use `on`. Although probably new method specifically for this use case would be better because you need a request from meet. You can use the `transport` request mechanism maybe. 

I'm not sure if it worth the effort! This is only my personal preference, so if you like it better than the current approach, let's do it, otherwise we can keep the current logic.

For the screen sharing I'd like to see this change! Unfortunately I'm not sure if it is not too late because of backward compatibility issues. I think this is the reason for using `window.JitsiMeetElectron` vs `window.JitsiMeetJS` which is our namespace for globals lately.

Reply:
Actually please hold on on this one because I'm reviewing the meet part right now and I may discard the comment :)